### ISSUE TITLE
Predict can't use `na.pass` when `newdata=NULL`

### ISSUE THREAD

**Issue by burchill** [lme4/lme4#505]
At least for my version of `lme4` (v.1.1.12), `na.action=na.pass` doesn't work when `predict.merMod` uses the original data set by default (i.e., `newdata=NULL`). It seems to automatically exclude whatever values have NAs in the fixed effects in the original data. The only way I found to get around this is to explicitly set `newdata` to the original data frame, which one should not be forced to do, and is very inconvenient.

I include code for a workable example below:

```r
library(purrr)
library(dplyr)
library(lme4)

set.seed(1)
effect <- 56

# Make by-subject slopes and intercepts
subj_df <- data.frame(
  Subj = factor(c(1:7)),
  SubjInts = rnorm(7, sd=5),
  SubjSlopes = rnorm(7, sd=5))

# Generate data
big_df <- purrr::map_dfr(
  1:100,
  ~mutate(subj_df, X=runif(7,max=300))) %>%
  mutate(Resp = X*effect + SubjInts + X*SubjSlopes + rnorm(Subj))

# Make a response `NA`
big_df[3,]$Resp <- NA

# Run the model
m <- lmer(Resp ~ X + (1 + X | Subj), data=big_df)

nrow(big_df) # 700
predict(m) %>% length() # 699
predict(m, na.action = na.pass) %>% length() # 699
predict(m, newdata = big_df) %>% length() # 700
```

**Comment by bbolker** [comment]
I agree that this should work the way you say it should, but: curious why specifying `newdata=original_data` is very inconvenient? Is this because you have a big data  frame that you could otherwise clear out of the workspace?

**Comment by burchill** [comment]
I'll be honest, it's not the most inconvenient thing ever (maybe I shouldn't have used 'very'), but here's my basic reasoning:

 * Like you said, if you've cleared out the original data frame, you either have to reload it/make a new variable to store it, or possibly do something like `predict(model_name, newdata = model.frame(model_name))`, which is messier and could fall into the other issues below
 * It's extra typing, which is annoying to do and makes code harder to read / format prettily
 * There are more chances/places where things can go wrong (both because that's inherently the case with more typing/code, and because something unexpected could affect the data frame variable you now have to pass in.

Not big issues really--more about aesthetics and coding philosophy, etc.
